packages/php-wasm/logger/src/lib/logger.ts

```diff
@@ -12,309 +15,159 @@ export type LogSeverity = 'Debug' | 'Info' | 'Warn' | 'Error' | 'Fatal';
 /**
  * Log prefix.
  */
-export type LogPrefix = 'Playground' | 'PHP-WASM';
+export type LogPrefix = 'PHP-WASM' | 'PHP' | 'JavaScript';
 
 /**
  * A logger for Playground.
  */
 export class Logger extends EventTarget {
 	public readonly fatalErrorEvent = 'playground-fatal-error';
 
-	/**
-	 * Log messages
-	 */
-	private logs: string[] = [];
-
-	/**
-	 * Whether the window events are connected.
-	 */
-	private windowConnected = false;
-
-	/**
-	 * The length of the last PHP log.
-	 */
-	private lastPHPLogLength = 0;
-
 	/**
 	 * Context data
 	 */
 	private context: Record<string, any> = {};
 
-	/**
-	 * The path to the error log file.
-	 */
-	private errorLogPath = '/wordpress/wp-content/debug.log';
-
-	constructor(errorLogPath?: string) {
+	// constructor
+	constructor(
+		// Log handlers
+		private readonly handlers: Function[] = []
+	) {
 		super();
-		if (errorLogPath) {
-			this.errorLogPath = errorLogPath;
-		}
 	}
 
 	/**
-	 * Read the WordPress debug.log file and return its content.
-	 *
-	 * @param UniversalPHP playground instance
-	 * @returns string The content of the debug.log file
+	 * Get all logs.
+	 * @returns string[]
 	 */
-	private async getRequestPhpErrorLog(playground: UniversalPHP) {
-		if (!(await playground.fileExists(this.errorLogPath))) {
-			return '';
-		}
-		return await playground.readFileAsText(this.errorLogPath);
+	public getLogs(): string[] {
+		// TODO use handlers to get logs
+		return [...logs];
 	}
 
 	/**
-	 * Log Windows errors.
-	 *
-	 * @param ErrorEvent event
+	 * Add context data.
 	 */
-	private logWindowError(event: ErrorEvent) {
-		this.log(
-			`${event.message} in ${event.filename} on line ${event.lineno}:${event.colno}`,
-			'Error'
-		);
+	public addContext(data: Record<string, any>): void {
+		this.context = { ...this.context, ...data };
```

Comment:
When called many times over time, it will overwrite the previous contextual information, right? And if we add non-conflicting keys, the previous one will stick around and potentially confuse us with information that's no longer true. Why maintain a separate context object? I see it's used to store `numberOfOpenPlaygroundTabs`, could that be just a log message?

Reply:
I added this to store context data for the error reporter, with the idea that it would have more data over time. 
Context may be the wrong name. The goal of this was to be the environment state and used similarly to PHP info. 
Most of the environment data would be set once as it's not changing, but some like numberOfOpenPlaygroundTabs would be updated. 

The more I think of it, this shouldn't be stored in logs as they are a record in time and should be immutable. 

I can move this to logs, and we can reintroduce context later if we need it.

Reply:
Done [fca0706](https://github.com/WordPress/playground/pull/1264/commits/fca070662c0badb7056a6d47208aab3939ba1e62)